include/dlaf/matrix.h

```diff
@@ -103,6 +103,34 @@ class Matrix : public Matrix<const T, device> {
     return operator()(this->distribution().localTileIndex(index));
   }
 
+  /// Copy values from another matrix
+  ///
+  /// Given a matrix with the same geometries and distribution, this function submits tasks that will
+  /// perform the copy of each tile
+  template <class U, class = std::enable_if_t<std::is_same<T, std::remove_const_t<U>>::value>>
+  void copyFrom(Matrix<U, Device::CPU>& source) {
```

Comment:
What about making it a free function to allow usage with matrices and views (using templates)?